plugins/uefi-capsule/fu-uefi-device.c

```diff
@@ -395,23 +395,37 @@ GBytes *
 fu_uefi_device_fixup_firmware(FuUefiDevice *self, GBytes *fw, GError **error)
 {
 	FuUefiDevicePrivate *priv = GET_PRIVATE(self);
-	gsize fw_length;
-	const guint8 *data = g_bytes_get_data(fw, &fw_length);
+	gsize bufsz;
+	const guint8 *buf = g_bytes_get_data(fw, &bufsz);
 	g_autofree gchar *guid_new = NULL;
 
 	priv->missing_header = FALSE;
 
 	/* GUID is the first 16 bytes */
-	if (fw_length < sizeof(fwupd_guid_t)) {
+	if (bufsz < sizeof(fwupd_guid_t)) {
 		g_set_error_literal(error,
 				    FWUPD_ERROR,
 				    FWUPD_ERROR_INVALID_FILE,
 				    "Invalid payload");
 		return NULL;
 	}
-	guid_new = fwupd_guid_to_string((fwupd_guid_t *)data, FWUPD_GUID_FLAG_MIXED_ENDIAN);
+
+	/* check for HP firmware that has some weird logo header prefix */
+	if (!fu_device_has_private_flag(FU_DEVICE(self),
+					FU_UEFI_DEVICE_FLAG_NO_CAPSULE_HEADER_FIXUP)) {
+		guint8 hdr[5] = {0x0};
+		if (!fu_memcpy_safe(hdr, sizeof(hdr), 0x0, buf, bufsz, 0x0, sizeof(hdr), error))
+			return NULL;
+		if (memcmp(hdr, "$LOGO", 5) == 0) {
+			g_warning("auto-enabling NO_CAPSULE_HEADER_FIXUP, this HP system probably "
```

Comment:
don't assume it's an HP system in the warning

Reply:
I'm really hesitant to turn it on unconditionally too. Maybe we just do the warning and don't set `NO_CAPSULE_HEADER_FIXUP`?

Reply:
I've asked HP for more information about the $LOGO thing. It might not be HP specific etc.